examples/design/simple_reactor.py

```diff
@@ -0,0 +1,547 @@
+# This could be stored as a JSON file and read in but for simplicity it is all written
+# here.
+# Notice there are no 'global' parameters as neither of the components share a variable.
+
+# %%
+
+build_config = {
+    "params": {},
+    "Plasma": {
+        "Designer": {
+            "params": {
+                "R_0": {
+                    "value": 9.0,
+                    "unit": "m",
+                    "source": "Input",
+                    "long_name": "Major radius",
+                },
+                "z_0": {
+                    "value": 0.0,
+                    "unit": "m",
+                    "source": "Input",
+                    "long_name": "Reference vertical coordinate",
+                },
+                "A": {
+                    "value": 3.1,
+                    "unit": "dimensionless",
+                    "source": "Input",
+                    "long_name": "Aspect ratio",
+                },
+                "kappa_u": {
+                    "value": 1.6,
+                    "unit": "dimensionless",
+                    "source": "Input",
+                    "long_name": "Upper elongation",
+                },
+                "kappa_l": {
+                    "value": 1.8,
+                    "unit": "dimensionless",
+                    "source": "Input",
+                    "long_name": "Lower elongation",
+                },
+                "delta_u": {
+                    "value": 0.4,
+                    "unit": "dimensionless",
+                    "source": "Input",
+                    "long_name": "Upper triangularity",
+                },
+                "delta_l": {
+                    "value": 0.4,
+                    "unit": "dimensionless",
+                    "source": "Input",
+                    "long_name": "Lower triangularity",
+                },
+                "phi_neg_u": {"value": 0, "unit": "degree", "source": "Input"},
+                "phi_pos_u": {"value": 0, "unit": "degree", "source": "Input"},
+                "phi_neg_l": {"value": 0, "unit": "degree", "source": "Input"},
+                "phi_pos_l": {"value": 0, "unit": "degree", "source": "Input"},
+            },
+        },
+    },
+    "TF Coil": {
+        "Designer": {
+            "runmode": "run",
+            "param_class": "PrincetonD",
+        },
+        "Builder": {
+            "params": {
+                "tf_wp_width": {
+                    "value": 0.6,
+                    "unit": "m",
+                    "source": "Input",
+                    "long_name": "Width of TF coil winding pack",
+                },
+                "tf_wp_depth": {
+                    "value": 0.8,
+                    "unit": "m",
+                    "source": "Input",
+                    "long_name": "Depth of TF coil winding pack",
+                },
+            },
+        },
+    },
+}
+
+
+# %%[markdown]
+
+# Now we set up our ParamterFrames
+
+# %%
+
+# TODO improve build config manipulation
```

Comment:
Big +1 on this. If there isn't one, we should get an issue opened for this for the v1 milestone. There's probably a fair bit of discussion to have on how we do this manipulation in the most sensible way